hazelcast/src/main/java/com/hazelcast/instance/Node.java

```diff
@@ -259,6 +260,13 @@ private void initializeListeners(Config config) {
                 known = true;
             }
 
+            if (listener instanceof InternalMigrationListener) {
+                final InternalPartitionServiceImpl partitionService =
+                        (InternalPartitionServiceImpl) nodeEngine.getPartitionService();
+                partitionService.setInternalMigrationListener((InternalMigrationListener) listener);
```

Comment:
This is not user api or spi. It is only done for testing. We use InternalMigrationListener to execute a specific test scenario deterministically within our integration tests so we register only a single internal listener and perform the test with it.